Backend/app/user/user_router.py

```diff
@@ -117,3 +119,21 @@ async def generate_device_UID(user_email: str):
     result = user_service.generate_new_uid(user_email)
 
     return result
+
+
+@router.patch("/loan")
+async def add_librarian(user: UpdateUserRoleDTO, token=Depends(HTTPBearer())):
```

Comment:
This would allow any user to turn themselves into a librarian if they wanted to. You should verify that the token belongs to an already existing librarian before continuing with this endpoint. The 'UpdateUserRoleDTO` should contain a role and the user email (since it's a librarian trying to update another user's role). 

This creates the little annoyance of: how can a librarian be created if there's no already existing librarians? We can manually modify the database to create the first librarian for now (first register as a regular user and then update that user's role) and search for a better solution later.